src/Microsoft.AspNet.Cors.Core/CorsPolicyBuilder.cs

```diff
@@ -0,0 +1,194 @@
+        /// Adds the specified <paramref name="origins"/> to the policy.
+        /// </summary>
+        /// <param name="origins">The origins that are allowed.</param>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AddOrigins(params string[] origins)
+        {
+            foreach (var req in origins)
+            {
+                _policy.Origins.Add(req);
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the specified <paramref name="headers"/> to the policy.
+        /// </summary>
+        /// <param name="headers">The headers which need to be allowed in the request.</param>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AddHeaders(params string[] headers)
+        {
+            foreach (var req in headers)
+            {
+                _policy.Headers.Add(req);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the specified <paramref name="exposedHeaders"/> to the policy.
+        /// </summary>
+        /// <param name="exposedHeaders">The headers which need to be exposed to the client.</param>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AddExposedHeaders(params string[] exposedHeaders)
+        {
+            foreach (var req in exposedHeaders)
+            {
+                _policy.ExposedHeaders.Add(req);
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the specified <paramref name="methods"/> to the policy.
+        /// </summary>
+        /// <param name="methods">The methods which need to be added to the policy.</param>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AddMethods(params string[] methods)
+        {
+            foreach (var req in methods)
+            {
+                _policy.Methods.Add(req);
+            }
+
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the policy to allow credentials.
+        /// </summary>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AllowCredentials()
+        {
+            _policy.SupportsCredentials = true;
+            return this;
+        }
+
+        /// <summary>
+        /// Sets the policy to not allow credentials.
+        /// </summary>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder DisallowCredentials()
+        {
+            _policy.SupportsCredentials = false;
+            return this;
+        }
+
+        /// <summary>
+        /// Ensures that the policy allows any origin.
+        /// </summary>
+        /// <returns>The current policy builder</returns>
+        public CorsPolicyBuilder AllowAnyOrigin()
+        {
+            _policy.Origins.Clear();
+            _policy.Origins.Add(CorsConstants.AnyOrigin);
```

Comment:
I thought about this, but this did not seem to be a big problem, mainly because the user agent is not expected to send a value * for the origin, header, or method, which means if you have a policy which has ```foo.com```, ```bar.edu```, and ```*```, in the policy, we would treat ```*``` as just another value. Now we might argue and say if a user adds a ```*``` using ```AddOrigins``` should we clear the Origins and allow any origin... to me it seems like too much smartness.

Having said that I was thinking of adding some validation for CorsPolicy which would validate things like an origin of the form ```http://foo.com/bar``` are not valid, in addition to origins having a ```*``` value.